### ISSUE TITLE
Memphis-Kom el-Fakhry Archaeological Project (MKAP): desired features

### ISSUE THREAD

**Issue by luizaogs** [arch-kiosk/arch-kiosk-office#1980]
I just went through the system with Didi to show her how it works and what tools it already includes, and the following are things she noticed she would want to add right off the bat. If we meet to talk about the project she would of course elaborate, but this is just to keep track for now: 

- visualizing 3D models (both photogrammetry and DEM): I think we can already upload all kinds of files to the file repository, right? but she wanted to know if there is also a way to visualize them (not needed in FM, only in Kiosk)
- something like a photogrammetry analysis module, maybe under the specialists tab, where there would be separate records for each model with information on each marker/point used to make the model
- ground control points (GCPs) for the creation of DEMs (xyz coordinates) with an image attached to each one - maybe this can be added as a tab to the survey module
- for burial context type, adding the drawing of a skeleton (we have a file already) in which the specific bone(s) found can be colored in (similar to deposit soil particle sorting)

Generally, she thought it was incredibly detailed and she particularly loved the QR codes for sorting images :)

**Comment by urapadmin** [comment]
> * visualizing 3D models (both photogrammetry and DEM): I think we can already upload all kinds of files to the file repository, right? but she wanted to know if there is also a way to visualize them (not needed in FM, only in Kiosk)

That one is the hardest of them. Currently we discourage uploading photogrammetry sources to the file repository even as ZIP files because they are just huge. Kiosk will never have capacities like google drive. We simply cannot have such an immense infrastructure, so uploading gigabytes of almost unused stuff is too much of a waste of our resources. 

When it comes to uploading the result of photogrammetry, we do have the idea of archives with previews for a while now #1231 / #138. That is definitely a planned feature but without a timeline, yet. If we can make the photogrammetry software export a glTF/GLB 3D model we could use that one as the preview file for the model (that itself would be in a zip file). Possible but I don't think all that soon. I would want to learn about the goal, first. Why does Didi want to see the models in the file repository? It will never be a substitute for specialized photogrammetry viewer. The only advantage that I can think of is finding them as one finds images (and soon other data) in Kiosk. To achieve that including the preview model in the file repository would be reasonable.

**Comment by luizaogs** [comment]
Yes, I guess it’s a wish to integrate that data with the rest rather than necessarily to visualize it in Kiosk itself. I think.

**Comment by luizaogs** [comment]
Ok, I just asked her: it would be good to have the information about the markers/points in the recording itself (maybe in an analysis module as I mentioned above), but linking to a separata database (like Drive or Dropbox) where the photos of the models themselves are and then just having the finished products (maybe with previews) in the file repository sounds great.

**Comment by urapadmin** [comment]
Makes me very happy that we already have a stable acronym :o) MKAP. It is written in stone now: We have a MKAP folder in the shared kiosk folder on google drive.

**Comment by urapadmin** [comment]
> Ok, I just asked her: it would be good to have the information about the markers/points in the recording itself (maybe in an analysis module as I mentioned above), but linking to a separate database (like Drive or Dropbox) where the photos of the models themselves are and then just having the finished products (maybe with previews) in the file repository sounds great.

I can imagine such a feature happening.

**Comment by urapadmin** [comment]
> * for burial context type, adding the drawing of a skeleton (we have a file already) in which the specific bone(s) found can be colored in (similar to deposit soil particle sorting)

That's a cool analysis model we have been talking about already for years. I think it is also a good feature for the shop window. People love visual stuff. We should tackle that one with priority.

**Comment by luizaogs** [comment]
> > * for burial context type, adding the drawing of a skeleton (we have a file already) in which the specific bone(s) found can be colored in (similar to deposit soil particle sorting)
> 
> That's a cool analysis model we have been talking about already for years. I think it is also a good feature for the shop window. People love visual stuff. We should tackle that one with priority.

Didi also has a more detailed form for burial recording from a project she participated in Hermopolis, if you guys want it as a reference. I don't think she needs more than the drawing for MKAP itself, but I think I remember Laurel talking about wanting to expand recording of this stuff in the past -- so this would be just in case you guys want to add other things.

**Comment by urapadmin** [comment]
> * something like a photogrammetry analysis module, maybe under the specialists tab, where there would be separate records for each model with information on each marker/point used to make the model
> 
> * ground control points (GCPs) for the creation of DEMs (xyz coordinates) with an image attached to each one - maybe this can be added as a tab to the survey module

We should talk about these two. The latter sounds to me like a responsibility of the GIS system (makes me wonder how stable and maintained projects keep their GIS databases from season to season, though). However, does that mean we'd store coordinates and images that show those control points in the soil so that they can be found again if necessary? I need some education here. (Still can't even memorize what DEM stands for)

**Comment by urapadmin** [comment]
> > > * for burial context type, adding the drawing of a skeleton (we have a file already) in which the specific bone(s) found can be colored in (similar to deposit soil particle sorting)
> > 
> > 
> > That's a cool analysis model we have been talking about already for years. I think it is also a good feature for the shop window. People love visual stuff. We should tackle that one with priority.
> 
> Didi also has a more detailed form for burial recording from a project she participated in Hermopolis, if you guys want it as a reference. I don't think she needs more than the drawing for MKAP itself, but I think I remember Laurel talking about wanting to expand recording of this stuff in the past -- so this would be just in case you guys want to add other things.

Well, dump all material in the shared MKAP folder and make a new feature ticket for this one: #1982

**Comment by lbestock** [comment]
@luizaogs - can you please get us the skeleton file? I want to start playing with that. MKAP folder in the google drive is where to put it. And might as well put the Hermopolis stuff there, too. And Didi must be on GitHub already so can we add her to this project and these MKAP tickets?

**Comment by luizaogs** [comment]
I don't have access to the MKAP Drive, but can add her to it once I do (and she will then add files). And I'll let her know to create a Github account, or will add her here if she already has one.

**Comment by lbestock** [comment]
Added both you and Didi to the MKAP drive - you should have gotten a notification.

**Comment by luizaogs** [comment]
> > * something like a photogrammetry analysis module, maybe under the specialists tab, where there would be separate records for each model with information on each marker/point used to make the model
> > * ground control points (GCPs) for the creation of DEMs (xyz coordinates) with an image attached to each one - maybe this can be added as a tab to the survey module
> 
> We should talk about these two. The latter sounds to me like a responsibility of the GIS system (makes me wonder how stable and maintained projects keep their GIS databases from season to season, though). However, does that mean we'd store coordinates and images that show those control points in the soil so that they can be found again if necessary? I need some education here. (Still can't even memorize what DEM stands for)

She will explain this much better than I ever could, but I think the idea would be to keep everything together. The points would be downloaded from the device as CSV and then can be added to ArcGIS for analysis, but images are not attached to those (and the CSV coordinates would need a home, anyway, so that they are accessible to the team). But she will explain more on Tuesday.

**Comment by luizaogs** [comment]
> @luizaogs - can you please get us the skeleton file? I want to start playing with that. MKAP folder in the google drive is where to put it. And might as well put the Hermopolis stuff there, too. And Didi must be on GitHub already so can we add her to this project and these MKAP tickets?

Burial recording form (with skeleton file) is in the drive.

**Comment by lbestock** [comment]
Oh interesting - that is pure MOLAS. Is the MKAP recording strategy generally MOLAS? Naqada is and it might be useful to work on those in parallel.

**Comment by luizaogs** [comment]
I don't think so, that's just what was used in Hermopolis (that's the Hermopolis form).

**Comment by luizaogs** [comment]
I don't think I can add Didi to the project myself, so here is her username: @didi-elbehaedi (https://github.com/didi-elbehaedi).

**Comment by urapadmin** [comment]
> I don't think I can add Didi to the project myself, so here is her username: @didi-elbehaedi (https://github.com/didi-elbehaedi).

I invited her

**Comment by luizaogs** [comment]
I added her to the tickets.

**Comment by luizaogs** [comment]
This is what I meant in terms of a photogrammetry analysis module. There is no place in the recording to record this stuff currently, and it is needed (and would otherwise be recorded in a separate paper form or something). Apologies for the bad image.

![C0CCB7F0-439B-49E9-984C-D8F5F176585F](https://user-images.githubusercontent.com/67970402/224047527-d7809e2b-9893-4943-baac-de72ab59bf4d.jpeg)

**Comment by urapadmin** [comment]
that's paper 😱

**Comment by luizaogs** [comment]
Didi would have to record this on paper too in Memphis 👀

**Comment by didi-elbehaedi** [comment]
I can confirm that this is super helpful in the field! If it’s possible to add it, that would be amazing ☺️

**Comment by urapadmin** [comment]
@didi-elbehaedi - We talked about importing coordinates from CSV files. I guess I need to see what such a file looks like.
Can you provide an example file? And if there are different sources (RTK and GIS) it would help to have an example from both of them.

**Comment by didi-elbehaedi** [comment]
Hi Lutz! Here are two examples: one is the raw data downloaded from the RTK (or Total Station, or GNSS Trimble R12 - these outputs should all be the same) and the other is after I have gone in and added headings. Both include the point names in the first column, with coordinates in the last three columns (latitude, longitude, and elevation UTM). Let me know if this is understandable or if you need anything else!

[Sample GNSS Points_RAW.csv](https://github.com/arch-kiosk/arch-kiosk-office/files/11011575/Sample.GNSS.Points_RAW.csv)
[Sample GNSS Points.csv](https://github.com/arch-kiosk/arch-kiosk-office/files/11011576/Sample.GNSS.Points.csv)

**Comment by urapadmin** [comment]
> Hi Lutz! Here are two examples: one is the raw data downloaded from the RTK (or Total Station, or GNSS Trimble R12 - these outputs should all be the same) and the other is after I have gone in and added headings. Both include the point names in the first column, with coordinates in the last three columns (latitude, longitude, and elevation UTM). Let me know if this is understandable or if you need anything else!
> 
> [Sample GNSS Points_RAW.csv](https://github.com/arch-kiosk/arch-kiosk-office/files/11011575/Sample.GNSS.Points_RAW.csv) [Sample GNSS Points.csv](https://github.com/arch-kiosk/arch-kiosk-office/files/11011576/Sample.GNSS.Points.csv)

Thanks. It helps a lot to actually see what we are talking about.
- If I am not mistaken these coordinates only mean anything specific if one knows the geodetic framework they belong to, like the coordinate system, the ellipsoid, the projection etc. Where would that information be stored since it is not part of the points?
- Do the names of the points "mean" anything? What do they refer to? A locus? An object? 
- what do you do with these files? Import them in GIS?

**Comment by luizaogs** [comment]
Lutz, I am responding on Didi's behalf (we are talking about this so these are her answers) as she packs next to me:

1. The points are calculated based on the projection, and when you import them into ArcGIS Pro you are able to select the projection used (or coordinate system). The collecting system itself knows that it's in UTM (or what have you). In the end the projection etc. is not important for the recording itself, but rather for the analysis part in the geospatial software. So we are not sure if this is relevant for the recording, or if it could just be noted separately. Long story short, that information is implied in the coordinates taken but not explicit anywhere.

2. In these samples, these points were permanent survey markers -- during the survey, nails were secured in the ground and those nails have those same names. Of course what the names of the points mean will depend on what they are being used for. For photogrammetry, for instance, the coordinates would correspond to specific markers present in the photos. If we're taking ground control points (GCPs), the names would correspond to the order the points were taken in (so GCP 1, GCP 2, etc.).

3. Yes, the files can be stored in the GIS if they are csv or txt.

**Comment by urapadmin** [comment]
okay. I'll make it a list of illustrated coordinates and one day you guys show me what you do with it :o)

**Comment by luizaogs** [comment]
We just had a MKAP meeting where Kiosk featured prominently, thought you guys would like to know :)

**Comment by urapadmin** [comment]
always helpful 👀

**Comment by didi-elbehaedi** [comment]
Hi Lutz! Luiza mentioned that you like to use project logos, so here is ours!
![No Background Logo](https://github.com/arch-kiosk/arch-kiosk-office/assets/125821304/440ba079-a5cd-4303-9801-612828f1a8b4)

**Comment by urapadmin** [comment]
And Luiza is right! Finally a project with some style ...
I made a colour scheme for MKAP, too, so that all the testers have some variety for their eyes.
All now available on mkap.arch-kiosk.brown.edu.
